src/v/transform/logging/log_manager.cc

```diff
@@ -0,0 +1,294 @@
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "transform/logging/log_manager.h"
+
+#include "base/vassert.h"
+#include "config/configuration.h"
+#include "model/namespace.h"
+#include "random/simple_time_jitter.h"
+#include "strings/utf8.h"
+#include "transform/logging/io.h"
+#include "transform/logging/logger.h"
+
+#include <seastar/core/condition-variable.hh>
+#include <seastar/core/manual_clock.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/coroutine/as_future.hh>
+
+namespace transform::logging {
+namespace {
+using namespace std::chrono_literals;
+} // namespace
+
+namespace detail {
+
+template<typename ClockType>
+class flusher {
+    static constexpr ClockType::duration jitter_amt{50ms};
+    static constexpr int MAX_CONCURRENCY = 10;
+
+public:
+    flusher() = delete;
+    explicit flusher(
+      ss::abort_source* as,
+      transform::logging::client* c,
+      config::binding<std::chrono::milliseconds> fi)
+      : _as(as)
+      , _client(c)
+      , _interval_ms(std::move(fi))
+      , _jitter(_interval_ms(), jitter_amt) {
+        _interval_ms.watch([this]() {
+            _jitter = simple_time_jitter<ClockType>{_interval_ms(), jitter_amt};
+        });
+    }
+
+    template<typename BuffersT>
+    ss::future<> start(ss::gate& gate, BuffersT* bufs) {
+        ssx::spawn_with_gate(
+          gate, [this, bufs]() -> ss::future<> { return flush_loop(bufs); });
+        return ss::now();
+    }
+
+    void wakeup() { _wakeup_signal.signal(); }
+
+    ss::future<> stop() {
+        _wakeup_signal.broken();
+        return ss::now();
+    }
+
+    ss::future<> flush_loop(auto* bufs) {
+        while (!_as->abort_requested()) {
+            try {
+                // the duration overload passes now() + dur to the timepoint
+                // overload, but the now() calculation is tied to an underlying
+                // clock type, so it doesn't work with ss::manual_clock. the
+                // timepoint overload template has a clocktype parameter, so we
+                // use that one to get the behavior we want for testing
+                if constexpr (std::is_same_v<ClockType, ss::manual_clock>) {
+                    co_await _wakeup_signal.wait(
+                      ClockType::now() + _jitter.base_duration());
+                } else {
+                    co_await _wakeup_signal.wait(_jitter());
+                }
+            } catch (const ss::broken_condition_variable&) {
+                break;
+            } catch (const ss::condition_variable_timed_out&) {
+            }
+            co_await flush(bufs);
+        }
+        co_return;
+    }
+
+    ss::future<> flush(auto* bufs) {
+        size_t n_events = 0;
+        absl::flat_hash_map<model::partition_id, io::json_batches> batches{};
+
+        if (bufs == nullptr) {
+            vlog(tlg_log.error, "Missing buffers");
+            co_return;
+        }
```

Comment:
This should never happen right? Should we `vassert` this in start?

Reply:
yep, was just adding the assertions locally